packages/contracts/test/plugin/plugin-setup-processor.ts

```diff
@@ -467,82 +547,119 @@ describe('Plugin Setup Processor', function () {
         );
 
         await expect(
           psp.prepareUninstallation(
             targetDao.address,
             plugin,
-            pluginSetup,
-            pluginSetupMockRepoAddress,
+            setupV1.address,
+            repoUUPS.address,
             helpers,
             EMPTY_DATA
           )
         ).to.be.revertedWith(customError('SetupNotApplied'));
       });
 
       it('reverts if plugin uninstallation is already prepared', async () => {
-        const pluginSetupBad = pluginSetupV1MockBad.address;
-
-        const installData = ethers.utils.defaultAbiCoder.encode(
-          ['address'],
-          [AddressZero]
+        // prepare first uninstallation
+        await prepareUninstallation(
+          psp,
+          targetDao.address,
+          proxy,
+          setupV1.address,
+          repoUUPS.address,
+          helpersV1,
+          EMPTY_DATA
         );
-        const {plugin, helpers, prepareInstallpermissions} =
-          await prepareInstallation(
+
+        // prepare second uninstallation
+        await expect(
+          prepareUninstallation(
             psp,
             targetDao.address,
-            pluginSetupBad,
-            pluginSetupMockRepoAddress,
-            installData
-          );
+            proxy,
+            setupV1.address,
+            repoUUPS.address,
+            helpersV1,
+            EMPTY_DATA
+          )
+        ).to.be.revertedWith(customError('SetupAlreadyPrepared'));
+      });
 
-        await psp.applyInstallation(
+      it('returns the plugin, helpers, and permissions', async () => {
+        let returnedPluginAddress;
+        let returnedHelpers;
+        let uninstallPermissionsV1: PermissionOperation[];
+        ({
+          returnedPluginAddress: returnedPluginAddress,
+          returnedHelpers: returnedHelpers,
+          permissions: uninstallPermissionsV1,
+        } = await prepareUninstallation(
+          psp,
           targetDao.address,
-          pluginSetupBad,
-          pluginSetupMockRepoAddress,
-          plugin,
-          prepareInstallpermissions
-        );
+          proxy,
+          setupV1.address,
+          repoUUPS.address,
+          helpersV1,
+          EMPTY_DATA
+        ));
 
-        // prepare first uninstallation
-        const uninstallData = ethers.utils.defaultAbiCoder.encode(
-          ['bool'],
-          [false]
+        expect(returnedPluginAddress).to.equal(proxy);
+        expect(returnedHelpers).to.deep.equal(helpersV1);
+        expect(uninstallPermissionsV1).to.deep.equal(
+          mockPermissionsOperations(0, 1, Operation.Revoke).map(perm =>
+            Object.values(perm)
+          )
         );
+      });
 
-        await psp.prepareUninstallation(
-          targetDao.address,
-          plugin,
-          pluginSetupBad,
-          pluginSetupMockRepoAddress,
-          helpers,
-          uninstallData
-        );
+      it('prepares a UUPS upgradeable plugin uninstallation', async () => {
```

Comment:
ThIs test not needed(useful) due to the previous one if succeeds, already states that it doesn't revert.

Reply:
Amended